src/Nancy/IO/RequestStream.cs

```diff
@@ -358,9 +358,23 @@ private void ShiftStreamToFileStreamIfNecessary()
 
         private static FileStream CreateTemporaryFileStream()
         {
-            var filePath = Path.GetTempFileName();
-
-            return new FileStream(filePath, FileMode.Create, FileAccess.ReadWrite, FileShare.None, 8192, StaticConfiguration.AllowFileStreamUploadAsync);
+            try
+            {
+                var filePath = Path.GetTempFileName();
+
+                return new FileStream(
+                    filePath,
+                    FileMode.Create,
+                    FileAccess.ReadWrite,
+                    FileShare.None,
+                    8192,
+                    StaticConfiguration.AllowFileStreamUploadAsync);
+            }
+            catch (IOException ex)
+            {
+				throw new IOException(
+                        "Unable to create temporary file, the %TEMP% folder has more than 65535 items. Try clearing out the folder.", ex);

```

Comment:
Can the `IOException` be thrown for reasons _other_ than going over the 65535 items limit? Would this obscure those reasons?